src/main/java/com/google/codeu/servlets/ItemDataServlet.java

```diff
@@ -0,0 +1,61 @@
+package com.google.codeu.servlets;
+
+import com.google.codeu.data.Item;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.codeu.data.Datastore;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+/**
+ * Handles fetching postings
+ */
+@WebServlet("/item-data")
+public class ItemDataServlet extends HttpServlet {
+  private Datastore datastore;
+
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String user = request.getParameter("user");
+    Item postingData = datastore.getPosting(user);
+    if (postingData == null) {

```

Comment:
I'd say instead of handling the null case, throw an error in datastore.getPosting then catch it here.
Then you can log the error so we know about it, but still handle the case and make a new posting.

I could see issues debugging where someone is getting their data reset and has no idea why (but it's actually a new posting), etc.